src/product/index.ts

```diff
@@ -0,0 +1,77 @@
+import api from "../api";
+import { Product, Response } from "../types";
+import { hasError } from "../util";
+import { DataTransform } from 'node-json-transform'
+import { productTransformRule } from "../mappings/product";
+
+export async function fetchProducts(params: any): Promise<any | Response> {
+  let response = {} as Product[] | Response
+
+  const payload = {
+    "json": {
+      "params": {
+        "rows": params.viewSize,
+        "start": params.viewIndex
+      },
+      "query": "*:*",
+      "filter": `docType: PRODUCT`
+    } as any
+  }
+
+  const queryFields = params.queryFields ? params.queryFields : "productId productName internalName";
+
+  // checking that if the params has filters, and then adding the filter values in the payload filter
+  // for each key present in the params filters
+  if (params.filters) {
+    Object.keys(params.filters).map((key: any) => {
+      const filterValue = params.filters[key].value;
+      const filterOperator = params.filters[key].op;
```

Comment:
I think it should be assigned only when it is array

Reply:
Done Sir